src/main/scala/webservice/CromIamApiService.scala

```diff
@@ -14,6 +23,35 @@ trait SwaggerService extends SwaggerUiResourceHttpService {
 
 trait CromIamApiService extends Directives with SprayJsonSupport with DefaultJsonProtocol with RouteConcatenation {
 
+  implicit val system: ActorSystem
+  implicit def executor: ExecutionContextExecutor
+  implicit val materializer: ActorMaterializer
+
+  val config: Config
+  val cromwellPool: Flow[(HttpRequest, Unit), (Try[HttpResponse], Unit), HostConnectionPool]
+
+  // Header that akka http DSL api adds to a request when we recieve,  We get an warning in logs if we dont strip it out before sending the request to cromwell
+  // HTTP header ‘Timeout-Access: <function1>’ is not allowed in requests
+  val timeoutAccesssHeader = "Timeout-Access"
+
+  def cromIamRequestToCromwellRequest(httpRequest: HttpRequest) : HttpRequest = {
+    httpRequest.copy(uri = httpRequest.uri.withAuthority(config.getString("cromwell.interface"), config.getInt("cromwell.port"))).withHeaders(httpRequest.headers.filterNot(header => header.name == timeoutAccesssHeader))
+  }
+
+  // left in here for now in case we decide to not use akka streams to send requests in the future, can probably remove soon
+  def forwardToCromwellNoStreaming(httpRequest: HttpRequest) : Future[HttpResponse] = {
+    Http().singleRequest(cromIamRequestToCromwellRequest(httpRequest))
+  }
+
+  def cromwellResponseHandler(value: (Try[HttpResponse], Unit)): HttpResponse = value match {
+    case (Success(response), _) => response
+    case (Failure(failure), _) => HttpResponse(status = StatusCodes.InternalServerError, entity = failure.getMessage + " : failed to talk to cromwell.")
+  }
+
+  def forwardToCromwellStreaming(httpRequest: HttpRequest) = {
+    Source.single((cromIamRequestToCromwellRequest(httpRequest), ())).via(cromwellPool).runWith(Sink.head).map(cromwellResponseHandler)
```

Comment:
I'd suggest to use `Http.singleRequest`. It's faster and less complicated. Using the streaming pool interface makes only sense if you really supply it with a stream of requests. Otherwise, it's functionally equivalent to use `Http.singleRequest`.